CMakeLists.txt

```diff
@@ -13,6 +13,11 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
                            -Werror=float-conversion)
 endif()
 
+if (NOT WIN32)
+  # Make all executables position independent on Linux
+  set(CMAKE_EXE_LINKER_FLAGS -pie)
```

Comment:
Sorry, haven't seen this before. Better use `string(APPEND CMAKE_EXE_LINKER_FLAGS " -pie")`. Otherwise you might overwrite options the user has provided. Example: If I call `cmake -DCMAKE_EXE_LINKER_FLAGS="whatever"` I don't want this option to be overwritten.

Otherwise LGTM! Thanks for this!

Reply:
done.. also added comment explaining why both flags and CMAKE_POSITION_INDEPENDENT_CODE need to be set.